src/renderer/vt/XtermEngine.cpp

```diff
@@ -380,7 +380,7 @@ try
         // statement here.
 
         // Move the cursor to the bottom of the current viewport
-        const short bottom = _lastViewport.BottomInclusive();
+        const short bottom = _viewport.BottomInclusive();

```

Comment:
This is also a nice example of why we should let the actual engine have more control. The `_lastViewport` was a duplicate of `_viewport` in `renderer.cpp`. Now it doesn't need to be duplicated anymore.

Reply:
Are you sure about that? Renderer has no interface for signaling a _resize_, and it provides no information in `_CheckViewportAndScroll` to the render engines about what the original size was. A member that stores a "last known value" and compares it against the new one is not a _duplicate_. It is a design decision.

Of course, I expect that any refactoring that seeks to de-"duplicate" those takes that design need into account.

Reply:
This is updated via engine-> UpdateViewport. All the logic are largely intact. I’ll double-check it to make sure it works as expected later.

Reply:
>it provides no information in _CheckViewportAndScroll to the render engines about what the original size was

Before this PR, `renderer.cpp` stores `_viewport` using:

https://github.com/microsoft/terminal/blob/d1f152adcfba5e9dc211e9ec0658b2fa1cefcd3a/src/renderer/base/renderer.cpp#L402

The exact same value was passed to `VtEngine` using:

https://github.com/microsoft/terminal/blob/d1f152adcfba5e9dc211e9ec0658b2fa1cefcd3a/src/renderer/base/renderer.cpp#L399

Then in VtEngine it stores the same value to `_lastViewport`:

https://github.com/microsoft/terminal/blob/79115e20585fcaafb703427c10d1a3365ac54c7b/src/renderer/vt/state.cpp#L218-L220

After this PR, I've manually checked that the same logic is now moved into `RenderEngineBase::UpdateViewport`. There's only one `_viewport` inside `RenderEngineBase` now. The initialization and update logic remain intact.

---

Plus all the UT are green (!) 

![image](https://user-images.githubusercontent.com/4710575/126578882-8264e100-9cc2-4c58-97db-76f1ac9191aa.png)